### DISCUSSION TITLE
EDI X12 message validation

### DISCUSSION THREAD

**Discussion by sadikpasha06** [xlate/staedi#483] [General]
Hello Team ,  i need a  java code/ libraries to validate the EDI x12 files.
specifically the EDI X12  834 and 837  transaction sets. 
please help me on this.

**Comment by MikeEdgar** [comment]
Hi @sadikpasha06 , there are several samples in the tests and test resources that can help get you started. For example, this test [1] reads a simple 837 and validates it against a subset of an 837 [2]. You would need to provide your own 837 standard schema and implementation schemas (read about the difference here [3]).

[1] https://github.com/xlate/staedi/blob/master/src/test/java/io/xlate/edi/internal/stream/SegmentValidationTest.java#L837
[2] https://github.com/xlate/staedi/blob/master/src/test/resources/x12/005010X222/837_loop1000_only.xml
[3] https://github.com/xlate/staedi/wiki/Validation

**Reply by sadikpasha06** [reply]
thanks @MikeEdgar do you have  any sample EDI 837/ 834 validation files.

**Reply by MikeEdgar** [reply]
@sadikpasha06 yes, see link [2] in my reply. It's an abbreviated 837, so you'll need to add the remaining loops according to your implementation guide requirements.

That schema also includes the base "standard" schema: https://github.com/xlate/staedi/blob/master/src/test/resources/x12/005010/837.xml
